core/core/segment.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+from typing import Any, Dict, Optional, Text
+
+import analytics
+import structlog
+from rasa.core.brokers.broker import EventBroker
+from rasa.utils.endpoints import EndpointConfig
+
+logger = structlog.get_logger()
+
+SEGMENT_URL_DEFAULT = "https://api.segment.io"
+
+
+def on_segment_error(error, items):
+    logger.warning("segment_tracking_failed", error=error, items=items)
+
+
+##
+# Taken (and adapted) from: https://github.com/dialoguemd/beautiful-mind/blob/master/rasa-stack/rasa_stack/segment.py
+
+
+class SegmentSource(EventBroker):
+    """ Track events with Segment"""
+
+    def __init__(self, write_key: Optional[str] = None):
+
+        if write_key is None:
+            write_key = os.environ.get("SEGMENT_WRITE_KEY")
```

Comment:
I've notified Maxime about that secret/environment variable - it will be added soon.

Reply:
(I will rebase by branch once he has added the secret, and check with him that he sees the events in segment.io)

Reply:
Done